### PR TITLE
[5.7] Add replace function to collection

### PR DESCRIPTION
**Pull request by blaisedufrain**
Replace a single value in a collection by specifying a key and value.  

Returns the updated collection.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -1407,6 +1407,18 @@ public function reject($callback)
         });
     }
 
+    /**
+     * Replace a single item based on the specified key.
+     *
+     * @param $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: invalid syntax

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by jmarcher** [comment]
This has some "unexpected" behaviour.

If I have something like: 
`collect(['name' => 'taylor', 'laravel' => 'framework'])`

If I use `collect(['name' => 'taylor', 'laravel' => 'framework'])->replace('name', 'john')` I expect:

`['name' => 'john', 'laravel'=> 'framework']`

but the actual result is:

`['laravel'=> 'framework', 'name' => 'john']`

**Comment by staudenmeir** [comment]
How is `replace()` different than `put()`?

**Comment by taylorotwell** [comment]
I keep telling people: collections are for *collections of items*. There are not for single objects.

**Pull request closed without merging**